src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -221,21 +221,35 @@ await requestContext.SendResult(new GenerateDeployPlanRequestResult()
             }
         }
 
-        private async Task ExecuteOperation(DacFxOperation operation, DacFxParams parameters, string taskName, RequestContext<DacFxResult> requestContext)
+        private void ExecuteOperation(DacFxOperation operation, DacFxParams parameters, string taskName, RequestContext<DacFxResult> requestContext)
         {
-            SqlTask sqlTask = null;
-            TaskMetadata metadata = TaskMetadata.Create(parameters, taskName, operation, ConnectionServiceInstance);
-
-            // put appropriate database name since connection passed was to master
-            metadata.DatabaseName = parameters.DatabaseName;
+            Task.Run(async () =>
```

Comment:
we don't need await here?

Reply:
if you do await here, it will block the main handler thread.

Reply:
closed offline